docs/bigdata/README.md

```diff
@@ -35,27 +35,75 @@ Navigate to **Custom core-site** to configure MinIO parameters for `_s3a_` conne
-
-![s3a-config](https://github.com/minio/minio/blob/master/docs/bigdata/images/image4.png?raw=true "custom core-site s3a")
+```
+sudo pip install yq
+alias kv-pairify='xq ".configuration[]" | jq ".[]" | jq -r ".name + \"=\" + .value"'
+```
+
+Let's take for example a set of 12 compute nodes with an aggregate memory of *1.2TiB*, we need to do following settings for optimal results. Add the following optimal entries for _core-site.xml_ to configure _s3a_ with **MinIO**. Most important options here are
+
+```
+cat ${HADOOP_CONF_DIR}/core-site.xml | kv-pairify | grep "mapred"
+
+mapreduce.map.java.opts=-Xmx104000m # 104 GB per node for mappers
+mapreduce.reduce.java.opts=-Xmx104000m # 104 GB per node for reducers
+mapred.job.reuse.jvm.num.tasks=15 # reuse upto 15 JVMs
+mapred.maxthreads.generate.mapoutput=2 # num threads to write map outputs
+mapred.maxthreads.partition.closer=0 # asynchronous map flushers
+mapred.min.split.size=2560000 # minimum split size
+mapreduce.fileoutputcommitter.algorithm.version=2
+mapreduce.job.reduce.slowstart.completedmaps=0.99 # 99% map, then reduce
+mapreduce.map.java.opts=-Xmx104000m # 104 GB per node for mappers
+mapreduce.map.speculative=false # disable speculation for mapping
+mapreduce.reduce.java.opts=-Xmx104000m # 104 GB per node for reducers
+mapreduce.reduce.maxattempts=1 # do not retry on failure
+mapreduce.reduce.merge.inmem.threshold=10000 # minimum # merges in RAM
+mapreduce.reduce.shuffle.input.buffer.percent=0.9 # min % buffer in RAM
+mapreduce.reduce.shuffle.merge.percent=0.9 # minimum % merges in RAM
+mapreduce.reduce.shuffle.parallelcopies=3840 # num of parition copies
+mapreduce.reduce.speculative=false # disable speculation for reducing
+mapreduce.task.io.sort.factor=999 # threshold before writing to disk
+mapreduce.task.io.sort.mb=1000 # memory for sorting
+mapreduce.task.sort.spill.percent=0.9 # minimum % before spilling to disk
+mapreduce.task.timeout=1200000 # map/reduce task timeout
+mapreduce.tasktracker.reserved.physicalmemory.mb.low=0.9
+```
+
+S3A is the connector to use S3 and other S3-compatible object stores such as MinIO. MapReduce workloads typically interact with object stores in the same way they do with HDFS. These workloads rely on HDFS atomic rename functionality to complete writing data to the datastore. Object storage operations are atomic by nature and they do not require/implement rename API. The default S3A committer emulates renames through copy and delete APIs. This interaction pattern causes significant loss of performance because of the write amplification. *Netflix*, for example, developed two new staging committers - the Directory staging committer and the Partitioned staging committer - to take full advantage of native object storage operations. These committers do not require rename operation. The two staging committers were evaluated, along with another new addition called the Magic committer for benchmarking.
```

Comment:
```suggestion
S3A is the connector to use with S3 and other S3-compatible object stores such as MinIO. MapReduce workloads typically interact with object stores in the same way as they do with HDFS. These workloads rely on HDFS atomic rename functionality to complete writing data to the datastore. Object storage operations are atomic by nature and they do not require/implement rename API. The default S3A committer emulates renames through copy and delete APIs. This interaction pattern causes significant loss of performance because of the write amplification. *Netflix*, for example, developed two new staging committers - the Directory staging committer and the Partitioned staging committer - to take full advantage of native object storage operations. These committers do not require a rename operation. The two staging committers were evaluated, along with another new addition called the Magic committer for benchmarking.
```